src/libsystemd/sd-login/sd-login.c

```diff
@@ -503,6 +503,36 @@ _public_ int sd_uid_get_display(uid_t uid, char **session) {
         return 0;
 }
 
+_public_ int sd_uid_get_login_time(uid_t uid, uint64_t *usec) {
+        _cleanup_free_ char *p = NULL, *s = NULL, *rt = NULL;
+        usec_t t;
+        int r;
+
+        assert_return(usec, -EINVAL);
+
+        r = file_of_uid(uid, &p);
+        if (r < 0)
+                return r;
+
+        r = parse_env_file(NULL, p, "STATE", &s, "REALTIME", &rt);
+        if (r == -ENOENT)
+                return -ENXIO;
+        if (r < 0)
+                return r;
+        if (isempty(s) || isempty(rt))
+                return -EIO;
+
+        if (strcmp(s, "active") && strcmp(s, "online"))
```

Comment:
According to systemd's coding style, please use `streq` to compare whether two strings are equal.